samples: drivers: spi_flash: Update sample to use alias

benjaminbjornsson opened the pull request.

benjaminbjornsson committed:
boards: arm: arduino_nicla_sense_me: Add node-label to nor-flash node

Add node-label to nor-flash node to make it easier to reference elsewhere.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

MaureenHelm reviewed (commented):
Why did you define the alias in overlays instead of in the main board dts?

benjaminbjornsson commented:
> Why did you define the alias in overlays instead of in the main board dts?

Probably too big of a step without a push... 😄 I'll move it.

benjaminbjornsson committed:
samples: drivers: spi_flash: Update sample to use alias

Simplify driver by using DT_ALIAS instead of DT_LABEL.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson commented on boards/arm/actinius_icarus_bee/actinius_icarus_bee_common.dts:
```diff
@@ -76,6 +76,7 @@
 		bootloader-led0 = &green_pwm_led;
 		mcuboot-button0 = &button0;
 		mcuboot-led0 = &green_pwm_led;
+		spi-flash = &w25q64;
```
@fabiobaltieri you're right, I'll fix that, thanks!

benjaminbjornsson committed:
boards: Add alias to boards with spi-flash node

Add alias to boards with spi-flash to facilitate the
update of samples/drivers/spi_flash.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson commented:
Cannot re-request review for some reason :)

fabiobaltieri commented:
@anangl can you re-approve?
